### PR TITLE
Current iteration of AuthZ changes

### PR DESCRIPTION
**Pull request by HaoK**
IAuthorizationPolicy has IAuthorizationRequirements
AuthorizationPolicy has RequireClaims/RequireRoles/Require(IAuthorizationRequirements)
IAuthorizationPolicyHandler are the global policy handlers

cc @lodejard @blowdart @divega

### CODE REVIEW

**Comment by HaoK** [comment]
Updated with latest conversation with @loudej @lodejard 

cc @yishaigalatzer

File: samples/SocialSample/SocialSample.kproj [hunk 1]
```diff
@@ -12,6 +12,7 @@
   </PropertyGroup>
   <PropertyGroup>
     <SchemaVersion>2.0</SchemaVersion>
+    <DevelopmentServerPort>13232</DevelopmentServerPort>
```
Comment (davidfowl) [inline samples/SocialSample/SocialSample.kproj]: zomg

File: src/Microsoft.AspNet.Security/AuthorizationOptions.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        private IDictionary<string, AuthorizationPolicy> PolicyMap { get; } = new Dictionary<string, AuthorizationPolicy>();
+
+        public void AddPolicy([NotNull] string name, [NotNull] AuthorizationPolicy policy)
+        {
+            PolicyMap[name] = policy;
+        }
+
+        public AuthorizationPolicy GetPolicy([NotNull] string name)
+        {
+            return PolicyMap.ContainsKey(name) ? PolicyMap[name] : null;
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Security/AuthorizationOptions.cs]: TryGetValue

File: src/Microsoft.AspNet.Security/Properties/debugSettings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "Profiles": []
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Security/Properties/debugSettings.json]: ignore

File: src/Microsoft.AspNet.Security/AuthorizationOptions.cs (hunk 2 above)
Comment (HaoK) [inline src/Microsoft.AspNet.Security/AuthorizationOptions.cs]: TryGetValue is just more characters no?

```
            AuthorizationPolicy result = null;
            return PolicyMap.TryGetValue(name, out result) ? result : null;

**Comment by lodejard** [comment]
:shipit:

**Comment by HaoK** [comment]
Note, one change Yishai suggested which I'm going to put in before I push, is that we move the DenyAnonymousRequirement and handler to Security since its not specific to MVC and has no depedencies.  Seemed ok so I'm going to move that code over here

**Comment by HaoK** [comment]
Merged 4a635835af1049679103fdca97716e814408addc

**Pull request closed without merging**